handlers/routesSelection.ts

```diff
@@ -199,7 +159,7 @@ export default function RoutesSelection({ flags }: SelectionConfig): Handler {
     const resp = await server(req, connInfo);
 
     // set cookie for the flags that has changed.
-    if (flagsThatShouldBeCookied.length > 0) {
+    if (flagsThatShouldBeCookied.length > 0 && resp.status < 300) { // errors and redirects have immutable headers
```

Comment:
for some reason errors and redirects have immutable headers, so we can't set them

Source: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Basic_concepts#guard

Notice that's not all errors but the Response.errors() and Response.redirect was. I've tested it manually setting the redirect status code which results in the same error, so I think it is internally set to immutable